sig-architecture/production-readiness.md

```diff
@@ -5,21 +5,90 @@ Kubernetes are observable, scalable and supportable, can be safely operated in
 production environments, and can be disabled or rolled back in the event they
 cause increased failures in production.
 
-More details may be found in the [PRR KEP].
+Production readiness reviews are done by a separate team, apart from the SIG
+leads (although SIG lead approval is needed as well, of course). It is useful to
+have the viewpoint of a team that is not as familiar with the intimate details
+of the SIG, but *is* familiar with Kubernetes and with operating Kubernetes in
+production. Experience through our dry runs in 1.19 and 1.20 have shown that
+this slightly "outsider" view helps identify otherwise missed items.
 
-## Status
+More background may be found in the [PRR KEP].
 
-As of 1.19, production readiness reviews are required, and are part of the KEP
-process. The PRR questionnaire previously found here has been incorporated into
-the [KEP template]. The template details the specific questions that must be
-answered, depending on the stage of the feature.
+## Status
 
 As of 1.21, PRRs are now blocking. PRR _approval_ is required for the enhancement
-to be part of the release.
+to be part of the release. This means that any KEPs targeting the release for any
+stage, and in - or moving to - `implementable` state, will require production
+readiness approval by the *Enhancements Freeze Date*.
+
+Note that some of the questions in the [KEP template] should be answered in both
+the KEP's README.md and the `kep.yaml`, in order to support automated checks on
+the PRR. The template points out these as needed.
+
+## Submitting a KEP for production readiness approval
+
+The KEP template production readiness questionnaire should be filled out by the
+KEP authors, and reviewed by the SIG leads. Once the leads are satisfied with
+both the overall KEP (i.e., it is ready to move to `implementable` state) and
+the PRR answers, the authors may request PRR approval:
+
+* Assign a PRR approver from the `prod-readiness-approvers` list in the
+  [OWNERS_ALIASES] file. This may be done earlier as well, to get early feedback
+  or just to let the approver know. Reach out on the `#prod-readiness` Slack
+  channel or just pick someone from the list.
+* Update the `kep.yaml`, setting the `stage`, `latest-milestone`, and the
+  `milestone` struct (which captures per-stage release versions).
+* Create a `prod-readiness/<sig>/<KEP number>.yaml` file, with the PRR
+  approver's GitHub handle for the specific stage
+* See this [example PRR approval request PR].
+
+The PRR approvers use the `kepctl` tool to identify
+all outstanding requests for PRR approval, and are responsible for providing
+timely feedback so that KEPs are not delayed. If your need is urgent it doesn't
```

Comment:
@deads2k @johnbelamaric - comment to ourselves
IIRC kepctl tool is based on the "prr-approver" field in the kep.yaml.
With our move towards "keps/prod-readiness/<kep-number>.yaml", I would like to remove prr-approver field (and it's also not reflecting the state-of-the-world, so we should probably update kepctl.

Reply:
so the assignment isn't made until the "keps/prod-readiness/.yaml" file is created? That's possible OK...but how do we do early reviews?

Reply:
I think that "prr-approver" in kep.yaml is now a bit redundant at this point.

I think for early reviews - we do them the same as we do:
- people who know who to assign - they will also have a PR with keps/prod-readiness/.yaml open
- people who don't know - we will be doing that based on query "no PR opened that adds that file"?

Reply:
ok, this is not something relevant for this PR, we will fix it when we change that

Reply:
Sure - that was more to us.